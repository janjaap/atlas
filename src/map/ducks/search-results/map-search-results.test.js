import reducer, {
  getMapSearchResults
} from './map-search-results';

it('sets the initial state', () => {
  expect(reducer(undefined, { type: 'UNKOWN' })).toMatchSnapshot();
});

it('handles search results request', () => {
  const action = getMapSearchResults({
    latitude: 52.3637006,
    longitude: 4.7943446
  }, { name: 'does not matter' });
  expect(reducer({}, action)).toMatchSnapshot();
});

it('handles search results success', () => {
  const action = {
    type: 'FETCH_MAP_SEARCH_RESULTS_SUCCESS',
    location: {
      latitude: 52.3637006,
      longitude: 4.7943446
    },
    mapSearchResults: [{ foo: 'bar' }] };
  expect(reducer({}, action)).toMatchSnapshot();
});
