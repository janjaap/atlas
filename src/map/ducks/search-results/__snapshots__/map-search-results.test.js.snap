// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles search results request 1`] = `
Object {
  "isLoading": true,
  "mapSearchResultsError": null,
}
`;

exports[`handles search results success 1`] = `
Object {
  "isLoading": false,
  "mapSearchResultsByLocation": Object {
    "52.3637006,4.7943446": Array [
      Object {
        "foo": "bar",
      },
    ],
  },
}
`;

exports[`sets the initial state 1`] = `
Object {
  "isLoading": false,
  "mapSearchResultsByLocation": Object {},
  "mapSearchResultsError": null,
}
`;
